corehq/util/celery_utils.py

```diff
@@ -186,3 +190,45 @@ def get_running_workers(timeout=10):
         worker_names.extend(worker_info.keys())
 
     return worker_names
+
+
+class LoadBasedAutoscaler(Autoscaler):
+    def _maybe_scale(self, req=None):
+        procs = self.processes
+        cur = min(self.qty, self.max_concurrency)
+
+        available_cpus = multiprocessing.cpu_count()
+        load_avgs = os.getloadavg()
+        max_avg = max(load_avgs[0], load_avgs[1])
+        normalized_load = max_avg / available_cpus
+
+        if cur > procs and normalized_load < 0.90:
+            self.scale_up(cur - procs)
+            return True
+        elif cur < procs:
+            self.scale_down((procs - cur) - self.min_concurrency)
+            return True
+        elif normalized_load > 0.90 and procs > self.min_concurrency:
+            # if load is to0 high trying scaling down 1 worker at a time.
+            # if we're already at minimum concurrency let's just ride it out
+            self.scale_down(1)
+            return True
+
+
+class OffPeakLoadBasedAutoscaler(LoadBasedAutoscaler):
+    def _is_off_peak(self):
+        now = datetime.utcnow().time()
+        if settings.OFF_PEAK_TIME:
+            if settings.OFF_PEAK_TIME[0] < now < settings.OFF_PEAK_TIME[1]:
+                return True
+
+        return False
+
+    def _maybe_scale(self, req=None):
+        procs = self.processes
+
+        if not self._is_off_peak() and procs > self.min_concurrency:
+            self.scale_down(1)
+            return True
```

Comment:
Won't this just oscillate between `min_concurrency + 1` and `min_concurrency` (assuming there are tasks in the queue)